### PR TITLE
[WIP] Blob Store Aggregator implementation

### PR DESCRIPTION
**Pull request by johngschulz**
First pass

### CODE REVIEW

File: geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    private List<BlobStoreConfiguration> configs;
+    private static Log log = LogFactory.getLog(org.geowebcache.storage.BlobStoreAggregator.class);
+    private ServiceInformation serviceInformation;
+    private GridSetBroker gridSetBroker;
+
+
+
+
+    public BlobStoreAggregator(GridSetBroker gridSetBroker, List<BlobStoreConfiguration> configs) {
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java]: As far as I can tell, GridSetBroker is just necessary for TileLayerDispatcher, not required for BlobStoreAggregator.

File: geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        int count = 0;
+        for (int i = 0; i < configs.size(); i++) {
+            BlobStoreConfiguration configuration = configs.get(i);
+            count += configuration.getBlobStoreCount();
+        }
+        return count;
+    }
+
+    public Set<String> getBlobStoreNames() {
+        Set<String> names = new HashSet<String>();
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java]: I'm not sure a set is viable here, as there is nothing enforcing name uniqueness across multiple blobStoreConfigurations

File: geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        config.renameBlobStore(oldName, newName);
+    }
+
+    /**
+     * Replaces the given blob store and returns the Blob Store's configuration, saving the configuration.
+     *
+     * @param bs The blob store to modify
+     * @throws IllegalArgumentException
+     */
+    public synchronized void modify(final BlobStoreInfo bs) throws IllegalArgumentException {
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java]: Reminder, lets try and keep the method names consistent between BlobStoreConfugration and BlobStoreAgregator; So in this case, `renameBlobStore` and `modifyBlobStore`

File: geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        //TODO: this won't work with GetCapabilitiesConfiguration
+        config.modifyBlobStore(bs);
+    }
+
+    public BlobStoreConfiguration getConfiguration(BlobStoreInfo bs) throws IllegalArgumentException {
+        Assert.notNull(bs, "blobStore is null");
+        return getConfiguration(bs.getName());
+    }
+
+    public BlobStoreConfiguration getConfiguration(final String blobStoreName) throws IllegalArgumentException {
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java]: Missing Javadocs here...

**Comment by tbarsballe** [comment]
I still see some public methods without javadocs, but everything else looks good.

**Comment by johngschulz** [comment]
Added the remaining javadocs, can merge when ready.

**Comment by tbarsballe** [comment]
Javadocs look good. Initialization stuff will probably need some modification once Kevin's changes get in; I'll merge this now and we can deal with the initialization later.

**Pull request merged**